hyperspy/drawing/markers.py

```diff
@@ -0,0 +1,903 @@
+    for i, ax in enumerate(signal_axes[::-1]):
+        # indexes need to be reversed
+        new_data[..., (-i - 1)] = ax.scale * peaks[..., i] + ax.offset
+    return new_data
+
+
+class Markers:
+
+    # The key defining the position, typically: `offsets`, `segments` or `verts`
+    _position_key = "offsets"
+    # For VerticalLines and HorizontalLines, the key to set is different from
+    # `_position_key`
+    _position_key_to_set = None
+
+    def __init__(
+        self,
+        collection,
+        offset_transform="data",
+        transform="display",
+        shift=None,
+        plot_on_signal=True,
+        name="",
+        ScalarMappable_array=None,
+        **kwargs,
+    ):
+        """
+        Create a set of markers using Matplotlib collections.
+
+        The markers are defined by a set of arugment required by the collections,
+        typically, ``offsets``, ``verts`` or ``segments`` will define their
+        positions.
+
+        To define a non-static marker any argument that can be set with the
+        :py:meth:`matplotlib.collections.Collection.set` method can be passed
+        as an array with `dtype=object` of the constructor and the same size as
+        the navigation axes of the a signal the markers will be added to.
+
+        Parameters
+        ----------
+        collection : matplotlib.collections or str
+            A Matplotlib collection to be initialized.
+        offset_transform, transform : str
+            Define the transformation used for the `offsets`. This only operates on the offset point so it won't
+            scale the size of the ``Path``.  It can be one of the following:
+            - ``"data"``: the offsets are defined in data coordinates and the ``ax.transData`` transformation is used.
+            - ``"relative"``: The offsets are defined in data coordinates in x and coordinates in y relative to the
+              data plotted. Only for 1D figure.
+            - ``"axes"``: the offsets are defined in axes coordinates and the ``ax.transAxes`` transformation is used.
+              (0, 0) is bottom left of the axes, and (1, 1) is top right of the axes.
+            - ``"xaxis"``: The offsets are defined in data coordinates in x and axes coordinates in y direction; use
+              :py:meth:`matplotlib.axes.Axes.get_xaxis_transform` transformation.
+            - ``"yaxis"``: The offsets are defined in data coordinates in y and axes coordinates in x direction; use
+              :py:meth:`matplotlib.axes.Axes.get_xaxis_transform` transformation..
+            - ``"display"``: the offsets are not transformed, i.e. are defined in the display coordinate system.
+              (0, 0) is the bottom left of the window, and (width, height) is top right of the output in "display units"
+              :py:class:`matplotlib.transforms.IndentityTransform`.
+        shift : None or float
+            Shift in matplotlib ``"axes"`` coordinate system. Must be a value
```

Comment:
This is for the "relative" case, right? It would be useful to elaborate what it does? For example, it seems that it can take an array of float and change the start and the end of the EELS lines markers? Is `shift` a good name?

Reply:
Yea this is for the relative case.  The idea is that the relative markers don't work great when you want to put a label right above a point.  It's better to shift thing by some constant amount relative to the axis.


```suggestion
            Only for ``offset_transform``="relative".  This applied a systematic shift in y to the offsets  in the matplotlib ``"axes"`` coordinate system. This provides a constant shift from the data for consistent labeling
```